core/config/locales/en.yml

```diff
@@ -476,8 +476,8 @@ en:
         one: Tax Rate
         other: Tax Rates
       spree/taxon:
-        one: Taxon
-        other: Taxons
+        one: Display Order
+        other: Display Order
```

Comment:
Safe from a technical or user perspective? I'm not sure of an alternative if it's code, but if it's the user you're worried about, we could use an interim 'Display Order (Taxon)' label that would be intuitive for familiar users and new users alike.